secant/Util/CombineSynchronizer.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  CombineSynchronizer.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 13.04.2022.
+//
+
+import Foundation
+import ZcashLightClientKit
+import Combine
+
+struct Balance: WalletBalance, Equatable {
+    var verified: Int64
+    var total: Int64
+}
+
+protocol CombineSynchronizer {
+    var synchronizer: SDKSynchronizer? { get }
+    var shieldedBalance: CurrentValueSubject<WalletBalance, Never> { get }
+    
+    func initWith(initializer: Initializer) throws
+    func prepare() throws
+    func start(retry: Bool) throws
+    func stop()
+    func updatePublishers()
+
+    func getTransparentAddress(account: Int) -> TransparentAddress?
+    func getShieldedAddress(account: Int) -> SaplingShieldedAddress?
+}
+
+extension CombineSynchronizer {
+    func start() throws {
+        try start(retry: false)
+    }
+
+    func getTransparentAddress() -> TransparentAddress? {
+        getTransparentAddress(account: 0)
+    }
+
+    func getShieldedAddress() -> SaplingShieldedAddress? {
+        getShieldedAddress(account: 0)
+    }
+}
+
+class LiveCombineSynchronizer: CombineSynchronizer {
+    private var cancellables: [AnyCancellable] = []
+    private(set) var synchronizer: SDKSynchronizer?
+    private(set) var shieldedBalance: CurrentValueSubject<WalletBalance, Never>
+
+    init() {
+        self.shieldedBalance = CurrentValueSubject<WalletBalance, Never>(Balance(verified: 0, total: 0))
+    }
+
+    func initWith(initializer: Initializer) throws {
+        synchronizer = try SDKSynchronizer(initializer: initializer)
+        
+        NotificationCenter.default.publisher(for: .synchronizerSynced)
+            .receive(on: DispatchQueue.main)
+            .sink(receiveValue: { [weak self] _ in
+                self?.updatePublishers()
+            })
+            .store(in: &cancellables)
+    }
+    
+    func prepare() throws {
```

Comment:
can `initWith(initializer: Initializer)` be dropped in favor of `prepare(with initializer: Initializer)`?

if prepare only runs `synchronizer?.prepare()` it will also save us from deciding what to do when a `prepare()` is called with no `synchronizer`.

Reply:
seems like it could be, I see no reason to split it, fixed